src/Our.Umbraco.Ditto/Web/Mvc/DittoView.cs

```diff
@@ -27,39 +28,52 @@ public abstract class DittoView<TViewModel>
         /// <param name="viewData">The view data.</param>
         protected override void SetViewData(ViewDataDictionary viewData)
         {
-            // If model is already ditto view model, use it
-            if (viewData.Model is DittoViewModel<TViewModel>)
+            // Gather the Ditto view model
+            var model = viewData.Model;
+
+            // If model is already Ditto view-model, use it
+            if (model is DittoViewModel<TViewModel>)
             {
                 base.SetViewData(viewData);
                 return;
             }
 
-            // Gather ditto view model properties
-            var model = viewData.Model;
+            // Gather any processor contexts
             var processorContexts = new List<DittoProcessorContext>();
 
+            // Check to see if this is a Ditto transfer model
             var transferModel = model as DittoTransferModel;
             if (transferModel != null)
             {
                 model = transferModel.Model;
                 processorContexts = transferModel.ProcessorContexts;
             }
 
-            // Check to see if we are a ditto view model at least and copy processors
+            // Get current content
+            var content = this.UmbracoContext.PublishedContentRequest != null
+                ? this.UmbracoContext.PublishedContentRequest.PublishedContent
+                : default(IPublishedContent);
+
+            // Get current culture
+            var culture = this.UmbracoContext.PublishedContentRequest != null
+                ? this.UmbracoContext.PublishedContentRequest.Culture
+                : CultureInfo.CurrentCulture;
+
+            // Check if the model is a Ditto base view-model; Use the assigned properties
             var baseDittoViewModel = model as BaseDittoViewModel;
             if (baseDittoViewModel != null)
             {
+                content = baseDittoViewModel.Content ?? content;
+                culture = baseDittoViewModel.CurrentCulture ?? culture;
```

Comment:
Yeah, I couldn't think of how they'd be different. I guess it's just because they were available on the `BaseDittoViewModel`?
I'm happy to remove that part.